libsplinter/src/admin/service/shared.rs

```diff
@@ -1017,6 +1023,94 @@ impl AdminServiceShared {
         }
     }
 
+    #[cfg(feature = "circuit-abandon")]
+    fn abandon_circuit(&mut self, circuit_id: &str) -> Result<(), ServiceError> {
+        // Verifying the circuit is able to be abandoned
+        let stored_circuit = self
+            .admin_store
+            .get_circuit(circuit_id)
+            .map_err(|err| {
+                ServiceError::UnableToHandleMessage(Box::new(AdminSharedError::SplinterStateError(
+                    format!("error occurred when trying to get circuit {}", err),
+                )))
+            })?
+            .ok_or_else(|| {
+                ServiceError::UnableToHandleMessage(Box::new(AdminSharedError::SplinterStateError(
+                    format!(
+                        "Received purged request for a circuit that does not exist: circuit id {}",
+                        circuit_id
+                    ),
+                )))
+            })?;
+
+        // send ABANDONED_CIRCUIT message to all other members' admin
+        // services
+        if let Some(ref network_sender) = self.network_sender {
+            let mut abandoned_circuit = AbandonedCircuit::new();
+            abandoned_circuit.set_circuit_id(circuit_id.to_string());
+            abandoned_circuit.set_member_node_id(self.node_id.clone());
+            let mut msg = AdminMessage::new();
+            msg.set_message_type(AdminMessage_Type::ABANDONED_CIRCUIT);
+            msg.set_abandoned_circuit(abandoned_circuit);
+
+            let envelope_bytes = msg.write_to_bytes().map_err(|err| {
+                ServiceError::UnableToHandleMessage(Box::new(MarshallingError::ProtobufError(err)))
+            })?;
+            for member in stored_circuit.members().iter() {
+                if member != &self.node_id {
+                    network_sender.send(&admin_service_id(member), &envelope_bytes)?;
+                }
+            }
+        }
+
+        if stored_circuit.circuit_status() != &StoreCircuitStatus::Active {
```

Comment:
This check should be before you send the messages to the other admin services

Reply:
Actually does this check even need to happen? here it should have already been validated in validate_abandon_circuit, also the error message is wrong. It should say "Attempting to abandon circuit that is not active"

Reply:
Looking again, no this check doesn't need to happen again since it should all be validated within the `validate_abandon_circuit` method

Reply:
At this point it should be validated, so I removed that check and kept that message where it is